app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -165,10 +179,13 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler {
             historyView.update(it)
         }
 
-        userHasHistory.observe(
-            viewLifecycleOwner,
-            historyView::updateEmptyState
-        )
+        requireContext().components.appStore.flowScoped(viewLifecycleOwner) { flow ->
```

Comment:
nit: Maybe we could pass the `appStore` to `HistoryFragment` allow easier testing in the future